src/couch_log/src/couch_log_formatter.erl

```diff
@@ -436,3 +471,31 @@ get_value(Key, List, Default) ->
 
 supervisor_name({local, Name}) -> Name;
 supervisor_name(Name) -> Name.
+
+format_meta(Meta) ->
+    %% https://www.rfc-editor.org/rfc/rfc5424.html#section-6.3
+    %% iut="3" eventSource="Application" eventID="1011"
+    string:join(maps:fold(fun(K, V, Acc) ->
+        [to_str(K, V) | Acc]
+    end, [], Meta), " ").
+
+%% passing complex terms as meta value is a mistake so we are going
+%% to eat it, because we cannot bubble up errors from logger
+%% Therefore following are not supported
+%% - lists
+%% - tuples
+%% - maps
+%% However we are not going to try to distinguish lists from string
+%% Atoms would be printed as strings
+to_str(K, _) when not (is_list(K) or is_atom(K)) ->
+    "";
+to_str(K, Term) when is_list(Term) ->
+    io_lib:format("~s=\"~s\"", [K, Term]);
+to_str(_K, Term) when is_tuple(Term) ->
+    "";
+to_str(_K, Term) when is_map(Term) ->
+    "";
+to_str(K, Term) when is_binary(Term) ->
+    io_lib:format("~s=\"~s\"", [K, Term]);
+to_str(K, Term) ->
```

Comment:
One question I had is whether we should limit precision of floats. I did a quick experiment to see whether "~p" limits the number of decimal places.

```erlang
6> io_lib:format("~p", [1.227773345398009300000000000000000000000000000000000000000000000000001]).
["1.2277733453980093"]
```

Reply:
Good call @iilyak. That seems fine to me, are you wanting to see more precise decimal formatting here?

Reply:
My note is for posterity so other reviewers do not spend time verifying this aspect. You can leave the implementation as is.